### ISSUE TITLE
Configuration of security policies

### ISSUE THREAD

**Issue by edugrasa** [IRATI/demonstrator#28]
Security policies have a bit specific syntax that cannot be covered with the current "policy" directive. In general security policies are specified like this in the configuration file:

````
 "securityManagerConfiguration": {
        "policySet": {
            "name": "default", 
            "version": "1"
        },
           "authSDUProtProfiles" : {
               "default" : {
                   "authPolicy" : {
                     "name" : "PSOC_authentication-ssh2",
                      "version" : "1",
                      "parameters" : [ {
                          "name" : "keyExchangeAlg",
                          "value" : "EDH"
                       }, {
                          "name" : "keystore",
                          "value" : "/creds"
                       }, {
                          "name" : "keystorePass",
                          "value" : "test"
                       } ]
                   },
                                 "encryptPolicy" : {
                 "name" : "default",
                 "version" : "1",
                 "parameters" : [ {
                   "name" : "encryptAlg",
                   "value" : "AES128"
                }, {
                   "name" : "macAlg",
                   "value" : "SHA256"
                }, {
                   "name" : "compressAlg",
                   "value" : "deflate"
                } ]
              },
              "TTLPolicy" : {
                 "name" : "default",
                 "version" : "1",
                 "parameters" : [ {
                    "name" : "initialValue",
                    "value" : "50"
                  } ]
                },
                "ErrorCheckPolicy" : {
                   "name" : "CRC32",
                   "version" : "1"
                }
               },
          "specific" : [ {
	       "underlyingDIF" : "regional.DIF",
               "authPolicy" : {
               	  "name" : "PSOC_authentication-password",
           	  "version" : "1",
           	  "parameters" : [ {
           	     "name" : "password",
           	     "value" : "kf05j.a1234.af0k"
           	  } ]
                },
                "TTLPolicy" : {
                   "name" : "default",
                   "version" : "1",
                   "parameters" : [ {
                      "name" : "initialValue",
                      "value" : "50"
                   } ]
                },
                "ErrorCheckPolicy" : {
                   "name" : "CRC32",
                   "version" : "1"
                }
           }]
         }
    }
````
The first part of the configuration can already be covered by the _policy_ directive
````
policy dcfabric.security-manager default
````

The problem are the other parts, which in fact are a container of a number of policies. The easiest way that I can think of supporting this is the one depicted in the example below, which should generate the above configuration file (using the policy directive as is, but playing with the component name). 

````
policy dcfabric security-manager.auth.default PSOC_authentication-ssh2 keyExchangeAlg=EDH keystore=/creds keystorePass=test
policy dcfabric security-manager.crypto.default default encryptAlg=AES128 macAlg=SHA256 compressAlg=deflate
policy dcfabric security-manager.ttl.default default initialValue=50
policy dcfabric security-manager.errorcheck.default CRC32
policy dcfabric security-manager.auth.regional PSOC_authentication-password password=kf05j.a1234.af0k
policy dcfabric security-manager.ttl.regional default initialValue=50
policy dcfabric security-manager.errorcheck.regional CRC32
````

**Comment by vmaffione** [comment]
I think it's a good idea. But there is bit of mismatch between the JSON and the policy declaration, isn't it? Did you mixed up "regional" and specific?

**Comment by edugrasa** [comment]
Specific is for a specific DIF that has to be named, therefore, if there is no "default" the configuration belongs to "specific" and the name is the DIF name. Or alternatively it could be like this:

````
policy dcfabric security-manager.ttl.specific.regional default initialValue=50
policy dcfabric security-manager.errorcheck.specific.regional CRC32
etc..
````

You have a better idea of what syntax is more convenient :)

**Comment by edugrasa** [comment]
Ok, what about this syntax?

````
policy dcfabric security-manager.auth.default PSOC_authentication-ssh2 keyExchangeAlg=EDH keystore=/creds keystorePass=test
policy dcfabric security-manager.crypto.default default encryptAlg=AES128 macAlg=SHA256 compressAlg=deflate
policy dcfabric security-manager.ttl.default default initialValue=50
policy dcfabric security-manager.errorcheck.default CRC32
policy dcfabric security-manager.auth.specific.regional PSOC_authentication-password password=kf05j.a1234.af0k
policy dcfabric security-manager.ttl.specific.regional default initialValue=50
policy dcfabric security-manager.errorcheck.specific.regional CRC32 
````

**Comment by vmaffione** [comment]
The "specific." seems redundant, we can remove it.
My understanding of the JSON syntax is that each entry of the "specific" array refers to a specific DIF (e.g. regional.DIF), while the default entry refers to all the DIFs. Am I right?

**Comment by edugrasa** [comment]
you got it :)

**Comment by vmaffione** [comment]
Fixed by #30
